test/evergreen.yml

```diff
@@ -200,7 +200,29 @@ functions:
         set -o errexit
         set -o verbose
         # Note: ${var} is used for expansions of evergreen variables, whereas $var are /bin/sh variables.
-        python_configured=$(grep PYTHON_EXECUTABLE: ./CMakeCache.txt | sed -e 's/.*=//')
+        echo "python_binary = '${python_binary}'"
+        echo "python_config_search_string = '${python_config_search_string}'"
+        echo "---- Start grep for ${python_config_search_string} ----"
+        grep "${python_config_search_string}": ./CMakeCache.txt
+        echo "---- End grep ----"
+        echo "---- Start grep for python ----"
+        grep python ./CMakeCache.txt
+        echo "---- End grep for python ----"
+        python_configured=$(grep ${python_config_search_string}: ./CMakeCache.txt | sed -e 's/.*=//')
+        echo "python_configured = '$python_configured'"
+
+        echo "============== Python Unit Test config check =============="
+        echo "python_binary                 = '${python_binary}'"
+        echo "                    version   = $(${python_binary} --version)"            
+        echo "python_config_search_string   = '${python_config_search_string}'"
+        echo "python_configured (via CMake) = '$python_configured'"
+        if [ "$python_configured" = '' ]; then
+            echo "                    No python configured via CMake"
+        else
+            echo "                    version   = $($python_configured --version)"    
+        fi
+        echo "==========================================================="
```

Comment:
I like the details dumped out here, which makes sense for local runs. However, I checked into one of the Evergreen task logs and found all those echo lines are printed twice there, one for the echo line itself and the other for the contents it echoed. Those lines intertwine with each other and make it a bit noisy. I guess Evergreen logs the shell code lines by default and wonder if there's a way to suppress them for this case.
```
[2023/06/07 01:21:46.592] # Note:  is used for expansions of evergreen variables, whereas $var are /bin/sh variables.
[2023/06/07 01:21:46.592] echo "python_binary = '/opt/homebrew/Frameworks/Python.framework/Versions/3.11/bin/python3.11'"
[2023/06/07 01:21:46.592] echo "python_config_search_string = '_Python3_EXECUTABLE'"
[2023/06/07 01:21:46.592] echo "---- Start grep for _Python3_EXECUTABLE ----"
[2023/06/07 01:21:46.592] python_binary = '/opt/homebrew/Frameworks/Python.framework/Versions/3.11/bin/python3.11'
[2023/06/07 01:21:46.592] python_config_search_string = '_Python3_EXECUTABLE'
[2023/06/07 01:21:46.592] ---- Start grep for _Python3_EXECUTABLE ----
[2023/06/07 01:21:46.594] _Python3_EXECUTABLE:INTERNAL=/opt/homebrew/Frameworks/Python.framework/Versions/3.11/bin/python3.11
[2023/06/07 01:21:46.594] grep "_Python3_EXECUTABLE": ./CMakeCache.txt
[2023/06/07 01:21:46.594] echo "---- End grep ----"
[2023/06/07 01:21:46.594] echo "---- Start grep for python ----"
[2023/06/07 01:21:46.596] ---- End grep ----
[2023/06/07 01:21:46.596] ---- Start grep for python ----
[2023/06/07 01:21:46.596] //Configure the python API
```

Reply:
I've resolved this by dramatically simplifying the code (and its testing) by creating a Python script to perform the version checking. It also outputs the details of the two versions (the specified Python binary and the one configured by CMake), and does so cleanly.

Reply:
For example:
`[2023/06/22 23:53:41.061] Python Version Analyser
[2023/06/22 23:53:41.061] =======================
[2023/06/22 23:53:41.061] This tool confirms that the Python version being used matches that configured in CMake.
[2023/06/22 23:53:41.061] Configuration:
[2023/06/22 23:53:41.061]   CMakeCache.txt file:      ./CMakeCache.txt
[2023/06/22 23:53:41.061]   Config search string:     _Python3_EXECUTABLE
[2023/06/22 23:53:41.061] Results:
[2023/06/22 23:53:41.061]   This Python:
[2023/06/22 23:53:41.061]       path  = /opt/homebrew/opt/python@3.11/bin/python3.11
[2023/06/22 23:53:41.061]       version  = Python 3.11.2
[2023/06/22 23:53:41.061]   CMake Python
[2023/06/22 23:53:41.061]       path  = /opt/homebrew/Frameworks/Python.framework/Versions/3.11/bin/python3.11
[2023/06/22 23:53:41.061]       version  = Python 3.11.2
[2023/06/22 23:53:41.061] Python versions (Python 3.11.2 and Python 3.11.2) are the same`

Reply:
That's neat.